rpcserver.go

```diff
@@ -839,6 +843,80 @@ func (r *rpcServer) ListUnspent(ctx context.Context,
 	return resp, nil
 }
 
+// PendingSweeps returns lists of on-chain outputs that lnd is currently
+// attempting to sweep within its central batching engine. Outputs with similar
+// fee rates are batched together in order to sweep them within a single
+// transaction. The fee rate of each sweeping transaction is determined by
+// taking the average fee rate of all the outputs it's trying to sweep.
+func (r *rpcServer) PendingSweeps(ctx context.Context,
+	in *lnrpc.PendingSweepsRequest) (*lnrpc.PendingSweepsResponse, error) {
+
+	// Retrieve all of the outputs the UtxoSweeper is currently trying to
+	// sweep.
+	pendingInputs, err := r.server.sweeper.PendingInputs()
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert them into their respective lnrpc format.
+	rpcPendingSweeps := make([]*lnrpc.PendingSweep, 0, len(pendingInputs))
+	for _, pendingInput := range pendingInputs {
+		var witnessType lnrpc.WitnessType
+		switch pendingInput.WitnessType() {
+		case input.CommitmentTimeLock:
+			witnessType = lnrpc.WitnessType_COMMITMENT_TIME_LOCK
+		case input.CommitmentNoDelay:
+			witnessType = lnrpc.WitnessType_COMMITMENT_NO_DELAY
+		case input.CommitmentRevoke:
+			witnessType = lnrpc.WitnessType_COMMITMENT_REVOKE
+		case input.HtlcOfferedRevoke:
+			witnessType = lnrpc.WitnessType_HTLC_OFFERED_REVOKE
+		case input.HtlcAcceptedRevoke:
+			witnessType = lnrpc.WitnessType_HTLC_ACCEPTED_REVOKE
+		case input.HtlcOfferedTimeoutSecondLevel:
+			witnessType = lnrpc.WitnessType_HTLC_OFFERED_TIMEOUT_SECOND_LEVEL
+		case input.HtlcAcceptedSuccessSecondLevel:
+			witnessType = lnrpc.WitnessType_HTLC_ACCEPTED_SUCCESS_SECOND_LEVEL
+		case input.HtlcOfferedRemoteTimeout:
+			witnessType = lnrpc.WitnessType_HTLC_OFFERED_REMOTE_TIMEOUT
+		case input.HtlcAcceptedRemoteSuccess:
+			witnessType = lnrpc.WitnessType_HTLC_ACCEPTED_REMOTE_SUCCESS
+		case input.HtlcSecondLevelRevoke:
+			witnessType = lnrpc.WitnessType_HTLC_SECOND_LEVEL_REVOKE
+		case input.WitnessKeyHash:
+			witnessType = lnrpc.WitnessType_WITNESS_KEY_HASH
+		case input.NestedWitnessKeyHash:
+			witnessType = lnrpc.WitnessType_NESTED_WITNESS_KEY_HASH
+		default:
+			rpcsLog.Warnf("Unhandled witness type %v for input %v",
+				pendingInput.WitnessType(),
+				pendingInput.OutPoint())
+		}
+
+		op := &lnrpc.OutPoint{
+			TxidBytes:   pendingInput.OutPoint().Hash[:],
+			OutputIndex: pendingInput.OutPoint().Index,
+		}
+		amountSat := uint32(pendingInput.SignDesc().Output.Value)
+		satPerByte := uint32(pendingInput.LastFeeRate.FeePerKVByte() / 1000)
```

Comment:
i'd be in favor exposing sat/kb instead to not lose the precision

Reply:
Seems better to keep the consistency with the rest of our APIs? Nothing else uses sat/kb to my knowledge.

Reply:
hmm yeah i guess most things are sat/b, we do have a couple things in sat/kw though